hudi-sync/hudi-bigquery-sync/src/main/java/org/apache/hudi/bigquery/HoodieBigQueryClient.java

```
@@ -0,0 +1,215 @@
+          ViewDefinition.newBuilder(query).setUseLegacySql(false).build();
+
+      bigquery.create(TableInfo.of(tableId, viewDefinition));
+      LOG.info("View created successfully");
+    } catch (BigQueryException e) {
+      throw new HoodieException("View was not created ", e);
+    }
+  }
+
+  @Override
+  public Map<String, String> getTableSchema(String tableName) {
+    // TODO: Implement automatic schema evolution when you add a new column.
+    return null;
+  }
+
+  @Override
+  public void addPartitionsToTable(final String tableName, final List<String> partitionsToAdd) {
+    // bigQuery discovers the new partitions automatically, so do nothing.
+  }
+
+  @Override
+  public void dropPartitionsToTable(final String tableName, final List<String> partitionsToDrop) {
+    // bigQuery discovers the new partitions automatically, so do nothing.
+  }
+
+  @Override
+  public boolean doesTableExist(final String tableName) {
+    // bigQuery table exists needs different set of arguments, so do nothing.
+    throw new UnsupportedOperationException("Not support doesTableExist yet.");
+  }
+
+  public boolean doesTableExist(final String projectId, final String datasetName, final String tableName) {
+    TableId tableId = TableId.of(projectId, datasetName, tableName);
+    return bigquery.getTable(tableId, BigQuery.TableOption.fields()) != null;
+  }
+
+  public boolean doesViewExist(final String projectId, final String datasetName, final String viewName) {
+    TableId tableId = TableId.of(projectId, datasetName, viewName);
+    return bigquery.getTable(tableId) != null;
+  }
+
+  @Override
+  public Option<String> getLastCommitTimeSynced(final String tableName) {
+    // bigQuery doesn't support tblproperties, so do nothing.
+    throw new UnsupportedOperationException("Not support getLastCommitTimeSynced yet.");
+  }
+
+  @Override
+  public void updateLastCommitTimeSynced(final String tableName) {
+    // bigQuery doesn't support tblproperties, so do nothing.
```

can we standardize on all unsupported APIs to throw `UnsupportedOperationException` ?

---

Good point, Resolved.